rest-api-spec/src/main/resources/rest-api-spec/api/fleet.search.json

```diff
@@ -0,0 +1,45 @@
+{
+  "fleet.search":{
+    "documentation":{
+      "url": null,
+      "description": "Search API where the search will only be executed after specified checkpoints are available due to a refresh. This API is design for internal use by the fleet server project."
+    },
+    "stability":"experimental",
+    "visibility":"public",
+    "headers":{
+      "accept": [ "application/json"],
+      "content_type": ["application/json"]
+    },
+    "url":{
+      "paths":[
+        {
+          "path":"/{index}/_fleet/_search",
+          "methods":[
+            "GET",
+            "POST"
+          ],
+          "parts":{
+            "index":{
+              "type":"string",
+              "description":"The name of the index."
+            }
+          }
+        }
+      ]
+    },
+    "params":{
+      "after_checkpoints_refreshed":{
```

Comment:
We typically use `wait_for` rather than `after`, would that be an option here? `wait_for_checkpoints_refresh`.